### DISCUSSION TITLE
Identify Supercharger from MQTT

### DISCUSSION THREAD

**Discussion by MikeBishop** [teslamate-org/teslamate#1933] [Q&A]
Looking at the Tesla API, it's easy to tell when the car is Supercharging:

        "fast_charger_present": true,
        "fast_charger_type": "Tesla",

However, the Teslamate MQTT doesn't seem to expose anything that identifies it clearly.  One can identify DCFC more generally -- voltage over 277V, for example -- but nothing identifies Superchargers specifically that I see.  Is there anything I'm missing?  If not, could it be added to the set of things exposed by MQTT, or is it a limitation of the Streaming API?

**Comment by adriankumpf** [comment]
AFAIK identifying superchargers did not always work reliably in the past based on the `fast_charger_type` / `fast_charger_brand`. Both fields are stored for every charging session in the `charges` table.  So you can check if it that approach would have worked  for your previous charging sessions.

In any case, adding new fields from from the Tesla API to the TeslaMate MQTT is relatively straight-forward.  We would just need to know what API fields you need exactly.
